--- /opt/ros/noetic/share/rosserial_arduino/arduino-cmake/cmake/Platform/Arduino.cmake	2021-03-05 20:38:32.252637890 -0500
+++ Arduino.cmake.fix	2021-03-05 20:45:02.137189204 -0500
@@ -991,8 +991,13 @@
 
             foreach(LIB_DEP ${LIB_DEPS})
                 setup_arduino_library(DEP_LIB_SRCS ${BOARD_ID} ${LIB_DEP} "${COMPILE_FLAGS}" "${LINK_FLAGS}")
-                list(APPEND LIB_TARGETS ${DEP_LIB_SRCS})
-                list(APPEND LIB_INCLUDES ${DEP_LIB_SRCS_INCLUDES})
+                # Do not link to this library. DEP_LIB_SRCS will always be only one entry
+                # if we are looking at the same library.
+                if(NOT DEP_LIB_SRCS STREQUAL TARGET_LIB_NAME)
+                    list(APPEND LIB_TARGETS ${DEP_LIB_SRCS})
+                    list(APPEND LIB_INCLUDES ${DEP_LIB_SRCS_INCLUDES})
+                endif()
+
             endforeach()
 
             if (LIB_INCLUDES)
